test/wasm_worker/proxied_function.js

```diff
@@ -1,7 +1,9 @@
 addToLibrary({
-  proxied_js_function__proxy: 'sync',
+  // Use a non-standard __proxy flag here.
+  // This is used, for example, in library_html5_webgl.j which uses
+  // sync_on_webgl_context_handle_thread.  With WASM_WORKERS we want the build
+  // to succeed rather than complain about the invalid proxy type.
+  proxied_js_function__proxy: 'xx_non_standard',
```

Comment:
Isn't this odd, though? It sounds like we're accepting weird input from users because of an internal implementation detail. Can wasm-workers mode set the proxying mode to some "wasm_workers" string and then have the code accept that iff in wasm-workers mode? That is, I'd hope some random string would still be an error in all modes.

Reply:
OK, we can do it that way instead.

I'm not sure how I would write a test though since I would just be erasing the problem.. but maybe that is fine.